PetShareTests/ShelterEndpointTests.cs

```diff
@@ -55,41 +77,72 @@ public async Task GetShouldFetchAllShelters()
         using var client = _testSetup.CreateFlurlClient().AllowAnyHttpStatus();
         var response = await client.Request("shelter").GetAsync();
         response.StatusCode.Should().Be(200);
-        var shelters = await response.GetJsonAsync<IEnumerable<ShelterResponse>>();
+        var shelters = await response.GetJsonAsync<PaginatedSheltersResponse>();
         shelters.Should().
-                 BeEquivalentTo(new[]
-                 {
-                     new ShelterResponse
-                     {
-                         Id = _shelter.Id,
-                         UserName = _shelter.UserName,
-                         FullShelterName = _shelter.FullShelterName,
-                         Email = _shelter.Email,
-                         PhoneNumber = _shelter.PhoneNumber,
-                         IsAuthorized = _shelter.IsAuthorized,
-                         Address = _shelter.Address
-                     }
-                 });
+                BeEquivalentTo(new PaginatedSheltersResponse
+                {
+                    Shelters = _shelters.Select(s => Shelter.FromEntity(s).ToResponse()).ToArray(),
+                    PageNumber = 0,
+                    Count = 2
+                });
+    }
+
+    [Fact]
+    public async Task GetShouldFailWithWrongPaginationParams()
+    {
+        using var client = _testSetup.CreateFlurlClient().AllowAnyHttpStatus();
+        var query = new PaginationQueryRequest
+        {
+            PageCount = 10,
+            PageNumber = 10,
+        };
+        var response = await client.Request("shelter").SetQueryParams(query).GetAsync();
+        response.StatusCode.Should().Be(StatusCodes.Status400BadRequest);
+    }
+
+    [Fact]
+    public async Task GetShouldFetchPaginatedFragmentOfShelters()
+    {
+        using var client = _testSetup.CreateFlurlClient().AllowAnyHttpStatus();
+        var query = new PaginationQueryRequest
+        {
+            PageCount = 1,
+            PageNumber = 0,
+        };
+        var response = await client.Request("shelter").SetQueryParams(query).GetAsync();
```

Comment:
We could pass query params as anonymous object here (like `.SetQueryParams(new { pageCount = 1, ... })`. This way if we (accidentally) change names in `PaginationQueryRequest` class, the test will start failing.

There's no point in changing that everywhere though (unless you have a lot of time).